pkg/handler/invites_test.go

```diff
@@ -0,0 +1,262 @@
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest(http.MethodGet, fmt.Sprintf("/api/v1/community/invites/config?guild_id=%s", tt.query), nil)
+
+			h.GetInviteTrackerConfig(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetInviteTrackerConfig] response mismatched")
+		})
+	}
+}
+
+func TestHandler_ConfigureInvites(t *testing.T) {
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, nil, nil, nil, nil)
+
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		args             interface{}
+		wantCode         int
+		wantResponsePath string
+	}{
+		{
+			name: "200_ok",
+			args: request.ConfigureInviteRequest{
+				LogChannel: "964773702476656701",
+				GuildID:    "962589711841525780",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/200-message-ok.json",
+		},
+		{
+			name: "400_invalid_json",
+			args: map[string]interface{}{
+				"log_channel": 964773702476656701,
+				"guild_id":    962589711841525780,
+			},
+			wantCode:         http.StatusBadRequest,
+			wantResponsePath: "testdata/guild_config_invite_trackers/400_invalid_json.json",
+		},
+		{
+			name: "400_missing_guild_id",
+			args: request.ConfigureInviteRequest{
+				LogChannel: "964773702476656701",
+			},
+			wantCode:         http.StatusBadRequest,
+			wantResponsePath: "testdata/400-missing-guildID.json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			body, err := json.Marshal(tt.args)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest(http.MethodPost, "/api/v1/community/invites/config", bytes.NewBuffer(body))
+
+			h.ConfigureInvites(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetInviteTrackerConfig] response mismatched")
+		})
+	}
+}
+
+func TestHandler_GetInvitesLeaderboard(t *testing.T) {
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, nil, nil, nil, nil)
+
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		args             string
+		wantCode         int
+		wantResponsePath string
+	}{
+		{
+			name:             "200_ok",
```

Comment:
can we add test for case 200 but list leaderboard empty ?

Reply:
You mean the test with guild does not have leaderboard? It's "200_guild_does_not_have_leaderboard" testcase